src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.ConvertHelpers.cs

```diff
@@ -195,6 +240,85 @@ public static partial class TensorPrimitives
             public static (Vector512<ushort> Lower, Vector512<ushort> Upper) Invoke(Vector512<byte> x) => Vector512.Widen(x);
         }
 
+        /// <summary>(uint)byte</summary>
+        private readonly struct WidenByteToUInt32Operator : IUnaryOneToFourOperator<byte, uint>
+        {
+            public static bool Vectorizable => true;
+
+            public static uint Invoke(byte x) => x;
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static (Vector128<uint>, Vector128<uint>, Vector128<uint>, Vector128<uint>) Invoke(Vector128<byte> x)
+            {
+                (Vector128<ushort> Lower, Vector128<ushort> Upper) ushorts = Vector128.Widen(x);
+                (Vector128<uint> Lower, Vector128<uint> Upper) uintsLower = Vector128.Widen(ushorts.Lower);
+                (Vector128<uint> Lower, Vector128<uint> Upper) uintsUpper = Vector128.Widen(ushorts.Upper);
+                return (uintsLower.Lower, uintsLower.Upper, uintsUpper.Lower, uintsUpper.Upper);
```

Comment:
@tannergooding, is there a better way to do this?

Reply:
Not in a cross platform fashion.

If you know that V512 is supported you can do a single instruction, it's just not portable.

This is probably a case we can recognize the pattern in the JIT and optimize it to do just that, long term, it should be pipelined and generally fast in the meantime

Reply:
Thanks. 

> If you know that V512 is supported you can do a single instruction

Which api is that?

Reply:
`Avx512F.ConvertToVector512Int32`, which takes in a `Vector128<byte>` and produces a `Vector512<int>`. You can then convert `int` to `float` using `Avx512F.ConvertToVector512Single`

Reply:
But it just produces one, presumably from the first quarter of the elements. What about the other elements? Do it does times with a shuffle for each?

Reply:
Not sure what you mean?

It takes in a `Vector128<byte>`, which contains `16x bytes`. It produces a `Vector512<int>` which contains `16x ints`. The vector size is changing to ensure you still have 16x elements and that all 16x are converted at once.

Right now you're doing:
```csharp
(var l, var u) = Vector128.Widen(x);
(var ll, var lu) = Vector128.Widen(l);
(var ul, var uu) = Vector128.Widen(u);
return (ll, lu, ul, uu);
```

This happens by emitting effectively:
```asm
vpmovzxbw xmm1, xmm0            ; zero extend lower half                 (this is completing an 8 to 16-bit conversion on input elements  0..7)
vpsrldq   xmm0, xmm0, 8         ; move upper half to lower half
vpmovzxbw xmm3, xmm0            ; zero extend lower half                 (this is completing an 8 to 16-bit conversion on input elements  8..15)

vpmovzxwd xmm0, xmm1            ; 16 to 32-bit zero extend lower half    (this is completing an 8 to 32-bit conversion on input elements  0..3)
vpsrldq   xmm1, xmm1, 8         ; move upper half to lower half
vpmovzxwd xmm1, xmm1            ; 16 to 32-bit zero extend lower half    (this is completing an 8 to 32-bit conversion on input elements  4..7)

vpmovzxwd xmm2, xmm3            ; 16 to 32-bit zero extend lower half    (this is completing an 8 to 32-bit conversion on input elements  8..11)
vpsrldq   xmm3, xmm3, 8         ; move upper half to lower half
vpmovzxwd xmm3, xmm3            ; 16 to 32-bit zero extend lower half    (this is completing an 8 to 32-bit conversion on input elements 12..15)
```

What I'm saying is you can instead do:
```csharp
Vector512<uint> r = Avx512F.ConvertToVector512UInt32(x);
var l = r.GetLower();
var u = r.GetUpper();
return (l.GetLower(), l.GetUpper(), u.GetLower(), u.GetUpper());
```

Which emits effectively:
```asm
vpmovzxbd     zmm0, xmm0        ; zero extend all elements               (this is doing an 8 to 32-bit conversion on input elements 0..15)
vextracti32x4 xmm1, xmm0, 1     ; extract v128                           (this gets output elements  4..7 into its own V128)
vextracti32x4 xmm2, xmm0, 2     ; extract v128                           (this gets output elements  8..11 into its own V128)
vextracti32x4 xmm3, xmm0, 3     ; extract v128                           (this gets output elements 12..15 into its own V128)
```

However, if we really wanted to optimize it we'd just process it as `V512<uint>` instead of as a tuple of `4x V128<uint>`, in which case we just emit `vpmovzxbd     zmm0, xmm0`

Reply:
> It takes in a Vector128<byte>, which contains 16x bytes. It produces a Vector512<uint> which contains 16x uints.

Hmm, ok, I must have been looking at one of the many other ConvertToVector methods, eg SSE's ConvertToVector128Int32 that takes a single Vector128 of byte and returns a single Vector138 of int

Reply:
I'll leave it as-is for now.